targets/simple_switch/simple_switch.cpp

```diff
@@ -670,7 +670,9 @@ SimpleSwitch::egress_thread(size_t worker_id) {
     phv->get_field("standard_metadata.egress_port").set(port);
 
     Field &f_egress_spec = phv->get_field("standard_metadata.egress_spec");
-    f_egress_spec.set(0);
+    // When egress_spec == drop_port the packet will be dropped, thus
+    // here we initialize egress_spec to a value different from drop_port.
+    f_egress_spec.set(drop_port + 1);
```

Comment:
Would we need a modulo here to ensure this is a legal port value?
Just a guess, I don't know if that would be needed.

Reply:
I believe egress_spec during egress processing, and after egress processing, is _not_ used as a port value at all.  It is simply compared to see if it is equal to, or not equal to, the drop port, at the end of egress.

If you want to be slightly safer, and more backwards-compatible with earlier behavior, you could instead use an assignment like this:
```
f_egress_spec.set((drop_port == 0) ? 1 : 0);
```

Reply:
the current implementation should be good enough
only requirement is that the initial egress_spec value be different from the chosen drop port